turbopack/crates/turbopack-core/src/resolve/alias_map.rs

```diff
@@ -492,20 +554,61 @@ where
                 match key {
                     AliasKey::Exact => {
                         if self.request.is_match(prefix) {
-                            return Some(AliasMatch::Exact(template.convert()));
+                            return Some(Ok(AliasMatch {
+                                prefix: prefix.clone(),
+                                key,
+                                output: template.convert(),
+                            }));
                         }
                     }
                     AliasKey::Wildcard { suffix } => {
-                        let mut remaining = self.request.clone();
-                        remaining.strip_prefix_len(prefix.len());
-                        let remaining_suffix = remaining.constant_suffix();
-                        if !remaining_suffix.ends_with(&**suffix) {
-                            continue;
+                        let is_match = if let Some(request) = self.request.as_constant_string() {
+                            // The request is a constant string, so the PatriciaMap lookup already
+                            // ensured that the prefix is matching the request.
+                            let remaining = &request[prefix.len()..];
+                            remaining.ends_with(&**suffix)
+                        } else if let Pattern::Concatenation(req) = self.request
+                            && let [
+                                Pattern::Constant(req_prefix),
+                                Pattern::Dynamic | Pattern::DynamicNoSlash,
+                            ] = req.as_slice()
+                        {
+                            // This and the following special case for commonly used subdir aliases
+                            // correspond to what Pattern::match_apply_template can achieve as well.
+
+                            // The request might be more specific than the mapping, e.g. for
+                            // `require('@/foo/' + dyn)` into a `@/*` mapping
+                            req_prefix.starts_with(&**prefix)
+                        } else if let Pattern::Concatenation(req) = self.request
+                            && let [
+                                Pattern::Constant(req_prefix),
+                                Pattern::Dynamic | Pattern::DynamicNoSlash,
+                                Pattern::Constant(req_suffix),
+                            ] = req.as_slice()
+                        {
+                            req_prefix.starts_with(&**prefix) && req_suffix.ends_with(&**suffix)
+                        } else {
+                            return Some(Err(anyhow::anyhow!(
+                                "complex patterns into wildcard exports fields are not \
+                                 implemented yet: '{}' into '{}*{}'",
```

Comment:
```suggestion
                                 implemented yet: {} into '{}*{}'",
```

I think it already has quotes